CMakeLists.txt

```diff
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
 cmake_policy(SET CMP0127 NEW)
 
 if(NOT DEFINED PROJECT_NAME)
-    set(CPP_INDIRECT_IS_SUBPROJECT ON)
+    set(CPP_INDIRECT_IS_NOT_SUBPROJECT ON)
```

Comment:
If `PROJECT_NAME` is not defined then the `project()` function has not been called up until this point.  As we only call this after this point then we can work out when another project has used this CMakeList.txt as a sub-component by detecting if this value is set.